public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx

```diff
@@ -2,86 +2,65 @@
 
-// interface State {
-//   query: LokiQuery;
-// }
-
 export class LokiQueryEditor extends PureComponent<Props> {
-  // state: State = {
-  //   query: this.props.query,
-  // };
-  //
-  // onRunQuery = () => {
-  //   const { query } = this.state;
-  //
-  //   this.props.onChange(query);
-  //   this.props.onRunQuery();
-  // };
-  //
-  // onFieldChange = (query: LokiQuery, override?) => {
-  //   this.setState({
-  //     query: {
-  //       ...this.state.query,
-  //       expr: query.expr,
-  //     },
-  //   });
-  // };
-  //
-  // onFormatChanged = (option: SelectableValue) => {
-  //   this.props.onChange({
-  //     ...this.state.query,
-  //     resultFormat: option.value,
-  //   });
-  // };
+  onToggleLive = () => {
+    const { query, onChange } = this.props;
+    onChange({
+      ...query,
+      live: !query.live,
+    });
+  };
 
   render() {
-    // const { query } = this.state;
-    // const { datasource } = this.props;
-    // const formatOptions: SelectableValue[] = [
-    //   { label: 'Time Series', value: 'time_series' },
-    //   { label: 'Table', value: 'table' },
-    // ];
-    //
-    // query.resultFormat = query.resultFormat || 'time_series';
-    // const currentFormat = formatOptions.find(item => item.value === query.resultFormat);
+    const { query, panelData, datasource, onChange, onRunQuery } = this.props;
+
+    let absolute: AbsoluteTimeRange;
+    if (panelData && panelData.request) {
+      const { range } = panelData.request;
+      absolute = {
+        from: range.from.valueOf(),
+        to: range.to.valueOf(),
+      };
+    } else {
+      absolute = {
+        from: Date.now() - 10000,
+        to: Date.now(),
+      };
+    }
 
     return (
       <div>
-        <div className="gf-form">
-          <div className="gf-form-label">
-            Loki is currently not supported as dashboard data source. We are working on it!
-          </div>
-        </div>
-        {/*
         <LokiQueryField
           datasource={datasource}
           query={query}
-          onQueryChange={this.onFieldChange}
-          onExecuteQuery={this.onRunQuery}
+          onChange={onChange}
+          onRunQuery={onRunQuery}
           history={[]}
+          panelData={panelData}
+          // ??? What do we need here?
+          syntaxLoaded={false}
+          onLoadOptions={() => {}}
+          logLabelOptions={[]}
+          syntax={[]}
+          datasourceStatus={DataSourceStatus.Connected}
+          absoluteRange={absolute}
         />
         <div className="gf-form-inline">
           <div className="gf-form">
-            <div className="gf-form-label">Format as</div>
-            <Select
-              isSearchable={false}
-              options={formatOptions}
-              onChange={this.onFormatChanged}
-              value={currentFormat}
-            />
+            <Switch label="Live" checked={!!query.live} onChange={this.onToggleLive} />
```

Comment:
Does this make sense here? I mean having a query set to live doesn't do anything when you use it in dashboard with some specific timerange does it? To me this seems like should be on the dashboard level same as time range and refresh interval. Basically I would just imagine a play/pause button somewhere in the UI that would change the whole dashboard to a live one (and just flip a flag on all the query targets)

Reply:
So right now if you set one query as live, it will just ignore the range and refresh rate and be always live?

Reply:
yea, we need to figure out in the coming months how streaming in dashboard is goint to work from UX perspective and how a user can understand that there are active streams, and a way to pause or start them.